### ISSUE TITLE
[web] Make search insensitive to differences in punctuation

### ISSUE THREAD

**Issue by hacketiwack** [owntone/owntone-server#1713]
**Current situation**
When searching for a song containing punctuation characters, the search string must match exactly the punctuation characters.

**Example**
Let's admit that you have a song named _L’Hymne à l’amour_ in your database and you want to search for it.
By entering the search term _l'hymne_ you won't get any result as the punctuation characters in the database are `’` (U+2019) instead of `'` (U+0027).
It would be awesome to be able to search with whatever punctuation characters.

For example, the search in Firefox takes that into account. Therefore, when searching for `’` or `'`, the results are the same.
Added to that, there are other punctuation characters that affect the search: e.g., `«`, `»`, `”`, `“`, `”`, `꞉`, `：`

**Comment by ejurgensen** [comment]
Yes, I guess this has relation to #1390, and I imagine the solution here is also some modification of the db collation. However, not sure how to actually do that, much of what is in done in that part of sqlext.c is black magic to me.

**Comment by hacketiwack** [comment]
@ejurgensen indeed, it is certainly related. I will try to find some more information on how to do that.
So far, my searches have given me no valuable results.
I'll keep you posted.

**Comment by hacketiwack** [comment]
I thought about this again, and I was wondering if using `uc_is_punct` could be used to ignore punctuation characters.

Maybe we could add a test [on this line](https://github.com/owntone/owntone-server/blob/9b12618b93983e9f1a58b27da45e4586277dd040/sqlext/sqlext.c#L396) to ignore punctuation characters like below.

```c
if ( !uc_is_punct(uString) && sqlite3Fts5UnicodeFold(uString, 1) != sqlite3Fts5UnicodeFold(uPattern, 1) )
```

However, I'm not sure what negative effects this would have.
